packages/messaging/src/controllers/sw-controller.ts

```diff
@@ -211,6 +217,9 @@ export class SwController implements FirebaseMessaging, FirebaseService {
         this.bgMessageHandler.next(payload);
       }
     }
+
+    // wait briefly to allow onBackgroundMessage to complete
+    await sleep(BACKGROUND_HANDLE_EXECUTION_TIME_LIMIT_MS);
```

Comment:
I'm sure you tested it. Just want to double check that we actually tested it in browser that it will prevent the warning to show.

Reply:
Also worth mentioning how we reached the number.